### PR TITLE
Refine the K8S kubespray installation guide

### LINKED ISSUE #34: k8s deployment guide with kubespray needs refinement
In [k8s deployment guide with kubespray](https://github.com/opea-project/docs/blob/6df3b09b07be8177eda6cab9b9516fa8121d7c51/guide/k8s_install_kubespray.md), I think we need the following improvement:

1. When gitclone kubespray, we should specify a release version, to avoid using unstable main branch.
2. When setting proxy, we should set additional_no_proxy instead of no_proxy, kubespray will generate the no_proxy based on configuration. When setting no_proxy, we overwrite kubespray's own mechanism to generate no_proxy, which is more prone to error.
3. We don't want kubespray to change the hostname to the value defined in ansible inventory file, which is unexpected.

### PR DESCRIPTION
**Pull request by lianhao**
- Use release version of kubespray

- Use addtional_no_proxy instead of no_proxy

- Avoid overwrite node hostname

- Fix other misspelling

Fixes #34.

### CODE REVIEW

**Review by lkk12014402 — APPROVED** [review]

**Review by daisy-ycguo — APPROVED** [review]
lgtm

File: guide/k8s_install_kubespray.md [hunk 1]
```diff
@@ -53,7 +55,7 @@ Ansible inventory defines the hosts and groups of hosts on which Ansible tasks a
 cp -r inventory/sample inventory/mycluster
 ```
 
-Edit your inventory file `inventory/mycluster/inventory.ini` to config the node name and IP address. The inventory file used in ths demo is as following:
+Edit your inventory file `inventory/mycluster/inventory.ini` to config the node name and IP address. The inventory file used in this demo is as following:
```
Comment (mkbhanda) [inline guide/k8s_install_kubespray.md:58]: s/following/follows:

**Review by mkbhanda — COMMENTED** [review]
I see the PR merged. If and when you get a chance would you address these minor suggestions.

**Comment by mkbhanda** [comment]
Also good to add minimum Python 3 version
for Ansible >= 2.16.4  python  3.10-3.12

**Comment by lianhao** [comment]
@mkbhanda Thanks for your suggestion. They're addressed in PR #36

**Pull request merged**